File: symfony-app/src/Entity/StockPharmacy.php
<?php

namespace App\Entity;

use App\Repository\StockPharmacyRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: StockPharmacyRepository::class)]
#[ORM\Table(name: 'stock_pharmacies', uniqueConstraints: [
    new ORM\UniqueConstraint(name: 'unique_stock', columns: ['pharmacie_id', 'medicament_id'])
])]
class StockPharmacy
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Pharmacy::class, inversedBy: 'stocks')]
    #[ORM\JoinColumn(name: 'pharmacie_id', nullable: false, onDelete: 'CASCADE')]
    private Pharmacy $pharmacie;

    #[ORM\ManyToOne(targetEntity: Medicament::class, inversedBy: 'stockPharmacies')]
    #[ORM\JoinColumn(name: 'medicament_id', nullable: false, onDelete: 'CASCADE')]
    private Medicament $medicament;

    #[ORM\Column(type: 'integer', options: ['default' => 0])]
    private int $quantite = 0;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2, nullable: true)]
    private ?string $prixVente = null;

    #[ORM\Column(type: 'date', nullable: true)]
    private ?\DateTimeInterface $dateExpiration = null;

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $createdAt;

    #[ORM\Column(type: 'datetime', options: ['default' => 'CURRENT_TIMESTAMP'])]
    private \DateTimeInterface $updatedAt;

    public function __construct()
    {
        $this->createdAt = new \DateTimeImmutable();
        $this->updatedAt = new \DateTimeImmutable();
    }

    public function getId(): ?int { return $this->id; }
    public function getPharmecie(): Pharmacy { return $this->pharmacie; }
    public function setPharmecie(Pharmacy $pharmacie): void { $this->pharmacie = $pharmacie; }
    public function getMedicament(): Medicament { return $this->medicament; }
    public function setMedicament(Medicament $medicament): void { $this->medicament = $medicament; }
    public function getQuantite(): int { return $this->quantite; }
    public function setQuantite(int $quantite): void { $this->quantite = $quantite; }
    public function getPrixVente(): ?string { return $this->prixVente; }
    public function setPrixVente(?string $prixVente): void { $this->prixVente = $prixVente; }
    public function getDateExpiration(): ?\DateTimeInterface { return $this->dateExpiration; }
    public function setDateExpiration(?\DateTimeInterface $dateExpiration): void { $this->dateExpiration = $dateExpiration; }
    public function getCreatedAt(): \DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(\DateTimeInterface $createdAt): void { $this->createdAt = $createdAt; }
    public function getUpdatedAt(): \DateTimeInterface { return $this->updatedAt; }
    public function setUpdatedAt(\DateTimeInterface $updatedAt): void { $this->updatedAt = $updatedAt; }
}

File: javafx-app/src/main/java/com/santea/model/StockPharmacy.java
package com.santea.model;

import java.time.LocalDateTime;

public class StockPharmacy {
    private Integer id;
    private Pharmacy pharmacie;
    private Medicament medicament;
    private Integer quantite;
    private String prixVente;
    private LocalDateTime dateExpiration;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public StockPharmacy() {
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Pharmacy getPharmacie() {
        return pharmacie;
    }

    public void setPharmacie(Pharmacy pharmacie) {
        this.pharmacie = pharmacie;
    }

    public Medicament getMedicament() {
        return medicament;
    }

    public void setMedicament(Medicament medicament) {
        this.medicament = medicament;
    }

    public Integer getQuantite() {
        return quantite;
    }

    public void setQuantite(Integer quantite) {
        this.quantite = quantite;
    }

    public String getPrixVente() {
        return prixVente;
    }

    public void setPrixVente(String prixVente) {
        this.prixVente = prixVente;
    }

    public LocalDateTime getDateExpiration() {
        return dateExpiration;
    }

    public void setDateExpiration(LocalDateTime dateExpiration) {
        this.dateExpiration = dateExpiration;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
}
